### ISSUE TITLE
Can't call lambda function for AWS S3 events

### ISSUE THREAD

**Issue by AbsolutOD** [Miserlou/Zappa#473]
I have been trying to create a lambda function that will parse cloudwatch logs for certain events and react on them.  I have just playing around to see what is in the event object coming from S3.  When I tail the logs, I can see the debug messages, but then I keep getting the following stack trace.

```python
[1479156641917] need more than 1 value to unpack: ValueError
Traceback (most recent call last):
  File "/var/task/handler.py", line 477, in lambda_handler
    return LambdaHandler.lambda_handler(event, context)
  File "/var/task/handler.py", line 189, in lambda_handler
    return handler.handler(event, context)
  File "/var/task/handler.py", line 349, in handler
    app_function = self.import_module_and_get_function(whole_function)
  File "/var/task/handler.py", line 179, in import_module_and_get_function
    module, function = whole_function.rsplit('.', 1)
ValueError: need more than 1 value to unpack
```

It has been pretty puzzling.  I can invoke the function from the command line, but all S3 events fail with the above stack trace.

https://github.com/Miserlou/Zappa/blob/master/zappa/handler.py#L343

I dug into the code and found why I am getting this error.  In the handler.py on line 343 it checks to see if the event has a 'Records' key and then calls `get_function_for_aws_event`.  The problem is that function returns the value in `event['s3']['configurationId']` which is returning the lambda's ARN (arn:aws:lambda:us-east-1:XXXXXXX:function:helloevent-dev).  The ARN causes the exception in `import_module_and_get_function`.

https://github.com/Miserlou/Zappa/blob/master/zappa/util.py#L99

When the S3 event is registered it names it after the lambda_arn.  The function `utility.get_event_source` takes the target_function as a param, but doesn't actually use it.  It sets the `funk.name = lambda_arn`.

https://github.com/Miserlou/Zappa/blob/master/zappa/util.py#L150

I think the the funk.name should be set to the target_function.  I am going to play around with it today.  I just wanted to post the issue in case someone had a better fix.

**Comment by bjinwright** [comment]
I can confirm this is happening.

**Comment by Miserlou** [comment]
Is this a regression?

**Comment by bjinwright** [comment]
Yep it happens in 0.27 as well

**Comment by AbsolutOD** [comment]
Yeah, I got a slightly different error.

```python
[1479240657613] need more than 1 value to unpack: ValueError
Traceback (most recent call last):
  File "/var/task/handler.py", line 476, in lambda_handler
    return LambdaHandler.lambda_handler(event, context)
  File "/var/task/handler.py", line 208, in lambda_handler
    raise ex
ValueError: need more than 1 value to unpack
```

**Comment by bjinwright** [comment]
@AbsolutOD What did you name your handler?

**Comment by AbsolutOD** [comment]
Here is the Events part of my zappa_settings.json file.

```json
        "events": [
            {
                "function": "events.retirement",
                "event_source": {
                    "arn":  "arn:aws:s3:::veracode-cloudtrail",
                    "events": [
                        "s3:ObjectCreated:*"
                    ]
                }
            }
        ]
```

**Comment by rileycrane** [comment]
The issue seems to be with how the `whole_function` is determined: https://github.com/Miserlou/Zappa/blob/887bded74cf20f989ebd59e07979df62890cc317/zappa/handler.py#L350 (edited)

It relies on a method `get_function_for_aws_event`  https://github.com/Miserlou/Zappa/blob/887bded74cf20f989ebd59e07979df62890cc317/zappa/handler.py#L263 which mistakenly grabs the `configurationId` from the Event payload
```        if 's3' in record:
            return record['s3']['configurationId']```
 instead of returning the `AWS_EVENT_MAPPING` for the function https://github.com/Miserlou/Zappa/blob/887bded74cf20f989ebd59e07979df62890cc317/zappa/handler.py#L273

**Comment by Miserlou** [comment]
Requesting @parroyo 's input on this one. It looks like we might have broken S3 events in order to support Kinesis events. Should be possible to handle both, of course, but we need to do it properly for all cases.

More importantly - this should never have slipped through the cracks. We need tests for every event source use case. These don't have to be placebo events, I think we can just use the example event sources that amazon details, although it'd be even better if we could capture actual events from the tail and then replay them during the test.

**Comment by Miserlou** [comment]
Fixed in 0.29.2
